src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+
+    /**
+     * SCENARIO #1
+     * 
+     * taddr -> taddrs
+     * 
+     * There are no zaddrs or joinsplits involved.
+     */
+    if (isPureTaddrOnlyTx) {
+        add_taddr_outputs_to_tx();
+        
+        CAmount funds = selectedUTXOAmount;
+        CAmount fundsSpent = t_outputs_total + minersFee;
+        CAmount change = funds - fundsSpent;
+        
+        if (change > 0) {
+            add_taddr_change_output_to_tx(change);
+
+            LogPrint("asyncrpc", "%s: transparent change in transaction output (amount=%s)\n",
+                    getId().substr(0, 10),
+                    FormatMoney(change, false)
+                    );
+        }
+        
+        Object obj;
+        obj.push_back(Pair("rawtxn", EncodeHexTx(tx_)));
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #1
+     */
+
+    
+    // Prepare raw transaction to handle JoinSplits
+    CMutableTransaction mtx(tx_);
+    mtx.nVersion = 2;
+    crypto_sign_keypair(joinSplitPubKey_.begin(), joinSplitPrivKey_);
+    mtx.joinSplitPubKey = joinSplitPubKey_;
+    tx_ = CTransaction(mtx);
+
+    // Copy zinputs and zoutputs to more flexible containers
+    std::deque<SendManyInputJSOP> zInputsDeque;
+    for (auto o : z_inputs_) {
+        zInputsDeque.push_back(o);
+    }
+    std::deque<SendManyRecipient> zOutputsDeque;
+    for (auto o : z_outputs_) {
+        zOutputsDeque.push_back(o);
+    }
+
+    
+    /**
+     * SCENARIO #2
+     * 
+     * taddr -> taddrs
+     *       -> zaddrs
+     * 
+     * Note: Consensus rule states that coinbase utxos can only be sent to a zaddr.
+     *       Any change over and above the amount specified by the user will be sent
+     *       to the same zaddr the user is sending funds to. 
+     */
+    if (isfromtaddr_) {
+        add_taddr_outputs_to_tx();
+        
+        CAmount funds = selectedUTXOAmount;
+        CAmount fundsSpent = t_outputs_total + minersFee + z_outputs_total;
+        CAmount change = funds - fundsSpent;
+        
+        // If there is a single zaddr and there are coinbase utxos, change goes to the zaddr.
+        if (change > 0) {
+            if (isSingleZaddrOutput && selectedUTXOCoinbase) {
+                std::string address = std::get<0>(zOutputsDeque.front());   
+                SendManyRecipient smr(address, change, std::string());
+                zOutputsDeque.push_back(smr);
+
+                LogPrint("asyncrpc", "%s: change from coinbase utxo is also sent to the recipient (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(change, false)
+                        );
+
+            } else if (!isSingleZaddrOutput && selectedUTXOCoinbase) {
+                // This should not happen and is not allowed
+                throw JSONRPCError(RPC_WALLET_ERROR, "Wallet selected Coinbase UTXOs as valid inputs when it should not have done");
```

Comment:
The code at the start of main_impl should ensure this never happens.  Changed throw to assert.